SCons/Job.py

```diff
@@ -396,19 +406,47 @@ def start(self):
                     else:
                         if task.needs_execute():
                             # dispatch task
-                            self.tp.put(task)
+                            self.thread_pool.put(task)
                             jobs += 1
+                            retired -= 1
+                            if retired == 0:
```

Comment:
I think this `if retired == 0` is redundant with the loop guard at the top, btw, so it should probably be dropped.

Reply:
yup. I believe your are correct!

Reply:
Fixed!